Sources/NIOHTTP2/InboundEventBuffer.swift

```diff
@@ -24,17 +24,27 @@ import NIOCore
 /// have always delivered all pending user events before we deliver a frame.
 /// Deliberately not threadsafe or `Sendable`.
 class InboundEventBuffer {
-    fileprivate var buffer: CircularBuffer<Any> = CircularBuffer(initialCapacity: 8)
+    fileprivate var buffer: CircularBuffer<BufferedHTTP2UserEvent> = CircularBuffer(initialCapacity: 8)
 
-    func pendingUserEvent(_ event: Any) {
+    func pendingUserEvent(_ event: BufferedHTTP2UserEvent) {
```

Comment:
To reduce the noise of this patch somewhat, it might be nicer to add individual `pendingUserEvent` functions for each of the types in the enum cases, and then the call sites don't have to change.